### DISCUSSION TITLE
Getting console errors for image which doesn't exist

### DISCUSSION THREAD

**Discussion by kempster** [alpinejs/alpine#3045] [7. Bugs (archived)]
In the context of some product cards, some items have 1 image and some have 2.

It's working fine on the front-end, but I'm getting console errors for each item which doesn't have a 2nd image.

Is this a bug, or am I missing something?

```
<div x-show="product.node.images.edges[1].node.originalSrc">
  <img x-bind:data-src="product.node.images.edges[0].node.originalSrc" src="">
  <img x-bind:data-src="product.node.images.edges[1].node.originalSrc" src="">
</div>
```

**Comment by SimoTod** [comment]
You can't access a variable that doesn't exist. X-show only applies display none to your div. You can use x-if instead

**Reply by kempster** [reply]
@SimoTod

Thanks - like this?

```
<div x-if="product.node.images.edges[1].node.originalSrc">
  <img x-bind:data-src="product.node.images.edges[0].node.originalSrc" src="">
  <img x-bind:data-src="product.node.images.edges[1].node.originalSrc" src="">
</div>
```

This shows a similar error, and also if I put the x-if on the image tag

**Reply by SimoTod** [reply]
Have a look at the documentation. X-if uses template tags.
